runtime/lua/vim/_editor.lua

```diff
@@ -266,8 +266,26 @@ do
     if undo_started then
       vim.api.nvim_command('undojoin')
     end
+
+    local recording = vim.fn.reg_recording() -- Check if recording is currently happening
+    local function feed_paste_as_input(text, input_mode)
+      if input_mode == 'i' then
+        vim.api.nvim_feedkeys(vim.api.nvim_replace_termcodes(text, true, false, true), 'it', true)
+      else
+        vim.api.nvim_feedkeys(vim.api.nvim_replace_termcodes(text, true, false, true), 'nt', true)
+        vim.api.nvim_feedkeys('', 'x', true)
+      end
+    end
+
+    local text = table.concat(lines, "\n")
+
     if mode:find('^i') or mode:find('^n?t') then -- Insert mode or Terminal buffer
-      vim.api.nvim_put(lines, 'c', false, true)
+      if recording ~= '' then
+        -- Feed the pasted text as keypresses so it gets recorded
+        feed_paste_as_input(text, 'i')
+      else
+        vim.api.nvim_put(lines, 'c', false, true)
```

Comment:
Appreciate your effort here. Would you be up for trying to update `nvim_put` so that it does this internally? That would hopefully avoid the (severe) performance issues of `nvim_feedkeys`.

How does your current PR perform if you paste 100k lines of input? OTOH, I guess macros aren't designed to handle that...

Reply:
Yes, I'd definitely like to try to update `nvim_put` to make this work internally. Just to confirm, when the pasted content reach `nvim_put` they will be added to the macro register(if recording) from there. Not from `vim.paste` as in the current PR.
Also, it would be really helpful if you could point me to the docs that I can refer to(for let's say checking if recording is going on).
Thank you for pointing out for testing it with large inputs. I haven't tried it with large inputs but will surely do while updating it.